manifests/install.pp

```diff
@@ -28,14 +28,16 @@
   file { $teleport::extract_path:
     ensure => $directory_ensure,
   } ->
-  archive { $teleport::archive_path:
-    ensure        => $ensure,
-    extract       => true,
-    extract_path  => $teleport::extract_path,
-    source        => $teleport::archive_url,
-    creates       => "${teleport::extract_path}/teleport",
-    checksum      => $teleport::checksum,
-    checksum_type => $teleport::checksum_type,
+  exec { 'download-teleport-latest':
+    command => "/usr/bin/aws s3 --region=${::region} cp ${teleport::archive_url} ${teleport_current_path}",
+    unless => "/usr/bin/sha256sum ${teleport_current_path} | /bin/grep ${teleport::archive_sha256}"
```

Comment:
[minor] maybe add `-q` to the grep command?

Reply:
Updated 👍